RxMarbles/OperatorsTableViewModel.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  OperatorsTableViewModel.swift
+//  RxMarbles
+//
+//  Created by Asaf Baibekov on 23/11/2019.
+//  Copyright © 2019 AnjLab. All rights reserved.
+//
+
+import UIKit
+
+class OperatorsTableViewModel {
+
+	private(set) var numberOfSections: Int
+
+	private let sections: [Section]
+	private var filteredSections: [Section]
+
+	init() {
+		self.sections = [
+			Section(name: "Combining", rows: [.combineLatest, .concat, .merge, .startWith, .switchLatest, .withLatestFrom, .zip]),
+			Section(name: "Conditional", rows: [.amb, .skipUntil, .skipWhile, .skipWhileWithIndex, .takeUntil, .takeWhile, .takeWhileWithIndex]),
+			Section(name: "Creating",rows: [.empty, .interval, .just, .never, .of, .repeatElement, .throw, .timer]),
+			Section(name: "Error", rows: [.catchError, .catchErrorJustReturn, .retry]),
+			Section(name: "Filtering", rows: [.debounce, .distinctUntilChanged, .elementAt, .filter, .ignoreElements, .sample, .single, .skip, .skipDuration, .take, .takeDuration, .takeLast, .throttle]),
+			Section(name: "Mathematical", rows: [.reduce]),
+			Section(name: "Transforming", rows: [.buffer, .delaySubscription, .flatMap, .flatMapFirst, .flatMapLatest, .map, .mapWithIndex, .scan, .toArray]),
+			Section(name: "Utility", rows: [.timeout])
+		]
+		self.filteredSections = [Section]()
+		self.numberOfSections = self.sections.count
+	}
+}
+
+extension OperatorsTableViewModel {
+	func titleForHeader(in section: Int) -> String? {
+		return self.getModelSection(at: section).name
+	}
+	func numberOfRows(in section: Int) -> Int {
+		return self.getModelSection(at: section).rows.count
+	}
+}
+
+extension OperatorsTableViewModel {
+	func updateSearchResults(for searchController: UISearchController) {
+		guard searchController.isActive, let text = searchController.searchBar.text, !text.isEmpty else { self.reset(); return }
```

Comment:
seems like everything is passive in the view model except for this - e.g. for everything else you get things from VM to VC. But in this case you pass in a VC to the VM, which might break encapsulation. Is there any way you can just pass back the needed data to the VC so it can consume it?

Reply:
I can agree with you, but it means I'd need to pass (isActive: Bool, text: Strings) params instead. I think that logically i'm saying, you give me your search controller, and I would update the data for according to its state.
It's not that I'm updating the "View" of the search controller through the view model.